### PR TITLE
Added UserProfile service, dao, dto, and extra controller methods

### PR DESCRIPTION
**Pull request by kkty39**
## Description

Please include a summary of the change and which issue is fixed. Include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 

Please also list any relevant details for your test configuration.

- [ ] Unit Test
- [ ] Postman Test
- [ ] Integration Test
- [ ] End-to-end Test (Frontend + Backend)

## Checklist:

Before you submit your Pull Request, please check you have completed the following:

- [ ] I have added tests that prove that my feature works or my fix is effective.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Additional Notes

Add any other details about the PR here.
Tests will be done later, currently finished writing basic logic.

### CODE REVIEW

**Review by charlesZSW — CHANGES_REQUESTED** [review]

File: UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UpdateUserProfileResponse.java
[the remark this answers is not in the document]
Comment (kkty39) [reply UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UpdateUserProfileResponse.java:1]: added

File: UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UserProfileFieldResponse.java
[the remark this answers is not in the document]
Comment (kkty39) [reply UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UserProfileFieldResponse.java:1]: added

File: UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UserProfileResponse.java
[the remark this answers is not in the document]
Comment (kkty39) [reply UserService/src/main/java/com/beaconfire/userservice/dto/UserProfileResponse/UserProfileResponse.java:1]: added

File: UserService/src/main/java/com/beaconfire/userservice/service/UserProfileService.java
[the remark this answers is not in the document]
Comment (kkty39) [reply UserService/src/main/java/com/beaconfire/userservice/service/UserProfileService.java:1]: added

File: UserService/src/main/java/com/beaconfire/userservice/controller/UserProfileController.java
[the remark this answers is not in the document]
Comment (kkty39) [reply UserService/src/main/java/com/beaconfire/userservice/controller/UserProfileController.java:1]: added

**Pull request closed without merging**